uaclient/entitlements/base.py

```diff
@@ -138,12 +157,22 @@ def can_disable(self, silent: bool = False) -> bool:
             return False
         return True
 
-    def can_enable(self, silent: bool = False) -> bool:
+    def can_enable(
+        self, silent: bool = False, enable_by_default: bool = False
+    ) -> bool:
         """
         Report whether or not enabling is possible for the entitlement.
 
+        :param enable_by_default: If True, do not stop enable if service
+                                  is marked as beta
         :param silent: if True, suppress output
         """
+
+        # If the contract needs to enable a beta service by default,
+        # we will not stop it here
+        if not self.valid_service and not enable_by_default:
+            return False
```

Comment:
It looks like the `enable_by_default` case is already handled on line 481 of this file, so I don't think we need it here.

Reply:
And so I didn't realize you were going to move the beta check into can_enable here - thank you!

This is most of fixing #1588, the rest is moving this check down to be the last check in `can_enable` (so that the "already enabled" message gets printed if necessary) and making this beta check print the same message that it was before (`MESSAGE_INVALID_SERVICE_OP_FAILURE_TMPL`) when not silent - I can handle those pieces in my PR or you can add them here - either works for me!

Reply:
Ahhh yes you're right!